db/schema_tables.cc

```diff
@@ -1638,25 +1674,61 @@ static shared_ptr<cql3::functions::user_aggregate> create_aggregate(replica::dat
     if (initcond_str) {
         initcond = state_type->from_string(initcond_str.value());
     }
+    ::shared_ptr<cql3::functions::scalar_function> reduce_func = nullptr;
+    if (rs && !rs->empty()) {
+        auto&& rrow = rs->row(0);
+        auto rfunc = rrow.get_nonnull<sstring>("reduce_func");
+
+        reduce_func = dynamic_pointer_cast<cql3::functions::scalar_function>(
+            cql3::functions::functions::find(cql3::functions::function_name{name.keyspace, rfunc}, {state_type, state_type}));
+
+        if (!reduce_func) {
+            throw std::runtime_error(format("Reduce function {} needed by aggregate {} not found", rfunc, name.name));
+        }
+    }
+
+    return ::make_shared<cql3::functions::user_aggregate>(name, initcond, std::move(state_func), std::move(reduce_func), std::move(final_func));
+}
+
+future<lw_shared_ptr<query::result_set>> 
```

Comment:
I think this should be an optional<row> to simplify use later on.

Reply:
This is a good idea but `row` doesn't have no-throw move constructor, so other option is to allocate pointer with `make_unique`, but then allocation is required.

Reply:
This function no longer exists, because of using before/after schema